lib/astFromSchema/index.js

```diff
@@ -0,0 +1,44 @@
+const sortASTs = require('../sortASTNodes')
+
+const createNameFilter = (names) => ({ name }) => !names.includes(name.value)
+
+function extractASTNodes (obj) {
+  let definitions = []
+  if (obj.astNode) {
+    definitions.push(obj.astNode)
+  }
+  if (obj.extensionASTNodes) {
+    definitions = definitions.concat(obj.extensionASTNodes)
+  }
+  return definitions
+}
+
+function astFromSchema (schema, opts) {
+  const {
+    filterDirectives = [],
+    filterTypes = []
+  } = opts
+
+  const handleTypesFilter = Array.isArray(filterTypes)
+    ? createNameFilter(filterTypes)
+    : filterTypes
+
+  const handleDirectivesFilter = Array.isArray(filterDirectives)
+    ? createNameFilter(filterDirectives)
+    : filterDirectives
```

Comment:
Is this sufficient to handle all the filter cases we need? What about interfaces, fields etc. The graphql tools docs outlines a lot more transformations. Is it something we may want to

a) flesh out more
b) leave out for now
c) find a way to integrate the existing graphql-tools to achieve the same thing.

Reply:
Sorry forgot to link the docs https://www.graphql-tools.com/docs/schema-wrapping#filtering

Reply:
The intention isn't to match graphql-tools functionality. We simply want to provide a means to filter out definitions from an AST. A specific use case would be removing federation directives from an SDL. The `types` field returned by `schema.toConfig()` includes interfaces along with objects, scalars, enums, unions, and inputs.